manta-crypto/src/merkle_tree/partial.rs

```diff
@@ -30,6 +30,239 @@ use core::{fmt::Debug, hash::Hash};
+    /// Generates a LeafMap from a 'Vec<LeafDigest>'
+    fn from_vec(leaf_digests: Vec<LeafDigest<C>>) -> Self;
+
+    /// Returns a vector with all leaf digests
+    #[inline]
+    fn leaf_digests(&self) -> Vec<LeafDigest<C>>
+    where
+        LeafDigest<C>: Clone,
+    {
+        (0..self.len())
+            .map(|x| self.get(x))
+            .filter(|x| match x {
+                None => false,
+                Some(_) => true,
+            })
+            .map(|x| x.unwrap().clone())
+            .collect()
+    }
+    /// Returns a vector with all marked leaf digests
+    #[inline]
+    fn marked_leaf_digests(&self) -> Vec<LeafDigest<C>>
+    where
+        LeafDigest<C>: Clone,
+    {
+        (0..self.len())
+            .filter(|&index| self.is_marked(index).unwrap_or(false))
+            .map(|x| self.get(x).unwrap().clone())
+            .collect()
+    }
+}
+
+/// Vector of leaf digests with markings
+#[derive(derivative::Derivative)]
+#[derivative(
+    Clone(bound = "LeafDigest<C>: Clone"),
+    Debug(bound = "LeafDigest<C>: Debug"),
+    Default(bound = "LeafDigest<C>: Default"),
+    Eq(bound = "LeafDigest<C>: Eq"),
+    Hash(bound = "LeafDigest<C>: Hash"),
+    PartialEq(bound = "LeafDigest<C>: PartialEq")
+)]
+pub struct LeafVec<C>
+where
+    C: Configuration + ?Sized,
+{
+    vec: Vec<(bool, LeafDigest<C>)>,
+}
+
+impl<C> LeafMap<C> for LeafVec<C>
+where
+    C: Configuration + ?Sized,
+    LeafDigest<C>: Clone + PartialEq,
+{
+    fn len(&self) -> usize {
+        self.vec.len()
+    }
+
+    fn get(&self, index: usize) -> Option<&LeafDigest<C>> {
+        let leaf_digest = &self.vec.get(index)?.1;
+        Some(leaf_digest)
+    }
+
+    fn current_index(&self) -> usize {
+        self.len() - 1
+    }
+
+    fn position(&self, leaf_digest: &LeafDigest<C>) -> Option<usize> {
+        self.vec.iter().position(|(_, l)| l == leaf_digest)
+    }
+
+    fn push(&mut self, leaf_digest: LeafDigest<C>) {
+        self.vec.push((false, leaf_digest));
+    }
+
+    fn from_vec(leaf_digests: Vec<LeafDigest<C>>) -> Self {
+        Self {
+            vec: leaf_digests
+                .iter()
+                .map(|x| (false, x.clone()))
+                .collect::<Vec<(bool, LeafDigest<C>)>>(),
+        }
+    }
+
+    fn mark(&mut self, index: usize) {
+        if let Some((b, _)) = self.vec.get_mut(index) {
+            *b = true
+        };
+    }
+
+    fn is_marked(&self, index: usize) -> Option<bool> {
+        let mark = self.vec.get(index)?.0;
+        Some(mark)
+    }
+
+    /// LeafVec does not implement leaf removal
+    fn remove(&mut self, _index: usize) -> bool {
+        false
+    }
+}
+
+/// Hash map of leaf digests.
+#[derive(derivative::Derivative)]
+#[derivative(
+    Clone(bound = "LeafDigest<C>: Clone"),
+    Debug(bound = "LeafDigest<C>: Debug"),
+    Default(bound = "LeafDigest<C>: Default"),
+    Eq(bound = "LeafDigest<C>: Eq"),
+    PartialEq(bound = "LeafDigest<C>: PartialEq")
+)]
+#[cfg(feature = "std")]
```

Comment:
The `cfg` should go above `derive` and should include the `doc` configs:

```suggestion
#[cfg(feature = "std")]
#[cfg_attr(doc_cfg, doc(cfg(feature = "std")))]
```